centreon/src/Core/ServiceGroup/Infrastructure/Repository/DbReadServiceGroupRepository.php

```diff
@@ -298,6 +487,318 @@ public function findByIds(int ...$serviceGroupIds): array
+
+        foreach ($bindValues as $key => $value) {
+            $statement->bindValue($key, $value, \PDO::PARAM_INT);
+        }
+
+        $statement->execute();
+
+        while (false !== ($hasAccessToAll = $statement->fetchColumn())) {
+            if (true === (bool) $hasAccessToAll) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param int[] $accessGroupIds
+     *
+     * @phpstan-param non-empty-array<int> $accessGroupIds
+     *
+     * @return bool
+     */
+    public function hasAccessToAllServiceCategories(array $accessGroupIds): bool
+    {
+        [$bindValues, $bindQuery] = $this->createMultipleBindQuery($accessGroupIds, ':access_group_id_');
+
+        $request = <<<SQL
+            SELECT 1
+            FROM `:db`.acl_resources_sc_relations arscr
+            INNER JOIN `:db`.acl_resources res
+                ON res.acl_res_id = arscr.acl_res_id
+            INNER JOIN `:db`.acl_res_group_relations argr
+                ON argr.acl_res_id = res.acl_res_id
+            INNER JOIN `:db`.acl_groups ag
+                ON ag.acl_group_id = argr.acl_group_id
+            WHERE ag.acl_group_id IN ({$bindQuery})
+            SQL;
+
+        $statement = $this->db->prepare($this->translateDbName($request));
+
+        foreach ($bindValues as $key => $value) {
+            $statement->bindValue($key, $value, \PDO::PARAM_INT);
+        }
+
+        $statement->execute();
+
+        return (bool) $statement->fetchColumn();
+    }
+
+    /**
+     * @param int[] $accessGroupIds
+     *
+     * @phpstan-param non-empty-array<int> $accessGroupIds
+     *
+     * @return bool
+     */
+    public function hasAccessToAllHostGroups(array $accessGroupIds): bool
+    {
+        [$bindValues, $bindQuery] = $this->createMultipleBindQuery($accessGroupIds, ':access_group_id_');
+
+        $request = <<<SQL
+            SELECT res.all_hostgroups
+            FROM `:db`.acl_resources res
+            INNER JOIN `:db`.acl_res_group_relations argr
+                ON argr.acl_res_id = res.acl_res_id
+            INNER JOIN `:db`.acl_groups ag
+                ON ag.acl_group_id = argr.acl_group_id
+            WHERE ag.acl_group_id IN ({$bindQuery})
+            SQL;
+
+        $statement = $this->db->prepare($this->translateDbName($request));
+
+        foreach ($bindValues as $key => $value) {
+            $statement->bindValue($key, $value, \PDO::PARAM_INT);
+        }
+
+        $statement->execute();
+
+        while (false !== ($hasAccessToAll = $statement->fetchColumn())) {
+            if (true === (bool) $hasAccessToAll) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param int[] $accessGroupIds
+     *
+     * @phpstan-param non-empty-array<int> $accessGroupIds
+     *
+     * @return bool
+     */
+    public function hasAccessToAllHostCategories(array $accessGroupIds): bool
```

Comment:
This method must be in the HostCategory repository or in a HostCategory repository trait..